internal/goflags/flags.go

```diff
@@ -0,0 +1,182 @@
+// Package goflags allows parsing go command invocations and storing their flags in a
+// CommandFlags structure. It also provides utilities to backtrack through the process stack to
+// find and parse the flags of the first parent go command found in the process hierarchy.
+package goflags
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"sync"
+
+	"github.com/datadog/orchestrion/internal/goproxy"
+	"github.com/shirou/gopsutil/v3/process"
+)
+
+// CommandFlags represents the flags provided to a go command invocation
+type CommandFlags struct {
+	Long    map[string]string
+	Short   map[string]struct{}
+	Unknown []string // flags we don't process but store anyway
+}
+
+var (
+	// specialFlags are long Go command flags that accept other flags as parameters
+	specialFlags = map[string]struct{}{
```

Comment:
What's the impact of this? Should these be merged with `longFlags`? It feels like they're handled identically... if the difference is only in what kind of values they get... this is not relevant here?